packages/hyperion-autologging/src/ALUIEventPublisher.ts

```diff
@@ -118,7 +118,7 @@ export function publish(options: InitOptions): void {
   trackInteractable(newEventsToPublish.map(ev => ev.eventName));
 
   let lastUIEvent: CurrentUIEvent | null;
-  let flowlet: ALFlowlet;
+  let flowletMap = new Map<UIEventConfig['eventName'], ALFlowlet>();
```

Comment:
Using map is one option. The `FlowletManager.pop` itself searches within the active flowlets for a matching flowlet to pop. That's why I required an argument in `.pop`.  The main problem is the "shared" flowlet variable between calls. Separating between calls by eventName should work most of the time, but just know that one other option is hang is on the target element itself. That the strategy I used for both XHR and Fetch in the `ALNetworkPublisher.ts`.

Reply:
I was thinking there could be a case where we are tracking multiple event types that could be occurring on the same target,  e.g. mousemove and click on the same element.